src/plugins/data/server/search/collectors/usage.ts

```diff
@@ -18,50 +18,28 @@ export interface SearchUsage {
 }
 
 export function usageProvider(core: CoreSetup): SearchUsage {
-  const getTracker = (eventType: keyof Usage) => {
-    return async (duration?: number) => {
-      const repository = await core
-        .getStartServices()
-        .then(([coreStart]) => coreStart.savedObjects.createInternalRepository());
-
-      let attributes: Usage;
-      let doesSavedObjectExist: boolean = true;
-
-      try {
-        const response = await repository.get<Usage>(SAVED_OBJECT_ID, SAVED_OBJECT_ID);
-        attributes = response.attributes;
-      } catch (e) {
-        doesSavedObjectExist = false;
-        attributes = {
-          successCount: 0,
-          errorCount: 0,
-          averageDuration: 0,
-        };
-      }
-
-      attributes[eventType]++;
-
-      // Only track the average duration for successful requests
-      if (eventType === 'successCount') {
-        attributes.averageDuration =
-          ((duration ?? 0) + (attributes.averageDuration ?? 0)) / (attributes.successCount ?? 1);
-      }
-
-      try {
-        if (doesSavedObjectExist) {
-          await repository.update(SAVED_OBJECT_ID, SAVED_OBJECT_ID, attributes);
-        } else {
-          await repository.create(SAVED_OBJECT_ID, attributes, { id: SAVED_OBJECT_ID });
-        }
-      } catch (e) {
-        // Version conflict error, swallow
-      }
-    };
-  };
+  const getRepository = once(async () => {
+    const [coreStart] = await core.getStartServices();
+    return coreStart.savedObjects.createInternalRepository();
+  });
 
   return {
-    trackError: () => getTracker('errorCount')(),
-    trackSuccess: getTracker('successCount'),
+    trackSuccess: async (duration: number) => {
+      const repository = await getRepository();
+      await repository.incrementCounter(SAVED_OBJECT_ID, SAVED_OBJECT_ID, [
+        { fieldName: 'successCount' },
```

Comment:
IIRC, we only migrate SO between versions id the data from one version to the next needs to be [persisted](https://www.elastic.co/guide/en/kibana/current/upgrade-migrations.html#upgrade-migrations) (i.e. end-user facing SO). Since telemetry-related SO is fetched once every 24 hours and indexed and _that_ hasn't changed, it doesn't really matter if we blow the SO object out the water and reinitialize a new one when the cluster upgrades. If you did want to persist the data, then I guess you'd need to add a migration. @Bamieh WDYT?

Reply:
I agree with tina. if a transformation is not possible of the old data simply using a new saved object id and dropping this one is fine. We can drop the existing data in the same savedobject type via `migrations`.

```
const migrate713 = (doc) => {
  return {
    ...doc,
    attributes: {},
  };
};
```

Reply:
I've updated this PR with a migration as suggested, @Bamieh could you take one more look? Thanks!